PlayChat/AppDelegate.swift

```diff
@@ -14,208 +14,216 @@
-    signOutButton.titleLabel?.textColor = UIColor.cyanColor()
-
-    signOutButton.addTarget(self, action: #selector(AppDelegate.signOut(_:)),
-                            forControlEvents: .TouchUpInside)
+
+    let signOutButton: UIButton = UIButton(
+      frame: CGRect(x: 5, y: height - 80, width: 55, height: 20))
+    signOutButton.setTitle(" << ", for: UIControlState())
+    signOutButton.titleLabel?.textColor = UIColor.cyan
+
+    signOutButton.addTarget(self, action: #selector(AppDelegate.signOut(_: )), for: .touchUpInside)
     channelView.view.addSubview(signOutButton)
-    
-    let textField:UITextField = UITextField(
-      frame: CGRectMake(60, height - 80, 300, 20))
+
+    let textField: UITextField = UITextField(
+      frame: CGRect(x: 60, y: height - 80, width: 300, height: 20))
     textField.attributedPlaceholder = NSAttributedString(
       string: "Enter your message",
-      attributes: [NSForegroundColorAttributeName: UIColor.lightGrayColor()])
-    textField.userInteractionEnabled = true
-    textField.textColor = UIColor.whiteColor()
-    textField.backgroundColor = UIColor.blackColor()
+      attributes: [NSAttributedStringKey.foregroundColor: UIColor.lightGray])
+    textField.isUserInteractionEnabled = true
+    textField.textColor = UIColor.white
+    textField.backgroundColor = UIColor.black
     textField.becomeFirstResponder()
     textField.delegate = self
     channelView.view.addSubview(textField)
-    
+
     return channelView
   }
-  
-  func application(application: UIApplication, openURL url: NSURL,
-                   options: [String: AnyObject]) -> Bool {
-    return GIDSignIn.sharedInstance().handleURL(url, sourceApplication:
-      options[UIApplicationOpenURLOptionsSourceApplicationKey] as? String,
-              annotation: options[UIApplicationOpenURLOptionsAnnotationKey])
+
+  func application(
+    _ application: UIApplication,
+    open url: URL,
+    options: [UIApplicationOpenURLOptionsKey: Any]
+  ) -> Bool {
+    return GIDSignIn.sharedInstance().handle(
+      url,
+      sourceApplication: options[UIApplicationOpenURLOptionsKey.sourceApplication] as? String,
+      annotation: [: ]
+    )
   }
-  func applicationWillResignActive(application: UIApplication) {}
-  func applicationDidEnterBackground(application: UIApplication) {}
-  func applicationWillEnterForeground(application: UIApplication) {}
-  func applicationDidBecomeActive(application: UIApplication) {}
-  func applicationWillTerminate(application: UIApplication) {}
-  
-  func signIn(signIn: GIDSignIn!, didSignInForUser user: GIDGoogleUser!,
-              withError error: NSError!) {
+  func applicationWillResignActive(_ application: UIApplication) {}
+  func applicationDidEnterBackground(_ application: UIApplication) {}
+  func applicationWillEnterForeground(_ application: UIApplication) {}
+  func applicationDidBecomeActive(_ application: UIApplication) {}
+  func applicationWillTerminate(_ application: UIApplication) {}
```

Comment:
These method stubs can just be removed entirely.

Reply:
Done.